school/src/main/scala/pl/edu/agh/school/algorithm/SchoolMovesController.scala

```diff
@@ -49,32 +30,42 @@ final class SchoolMovesController(bufferZone: TreeSet[(Int, Int)])(implicit conf
       if x != 0 && y != 0 && x != config.gridSize - 1 && y != config.gridSize - 1
     } {
       if (random.nextDouble() < config.spawnChance) {
-        grid.cells(x)(y) =
-          if (random.nextDouble() < config.foraminiferaSpawnChance) {
-            studentsCount += 1
-            CleanerAccessible.unapply(EmptyCell.Instance).withCleaner(config.foraminiferaStartEnergy, 0)
-          }
-          else {
-            teachersCount += 1
-            StudentAccessible.unapply(EmptyCell.Instance).withStudent(0)
-          }
+        val chance = random.nextDouble()
+        Math.abs(random.nextInt()) % 3 match {
+          case 0 =>
+            if (chance < config.cleanerSpawnChance) {
```

Comment:
You can move this `if` to `case 0` just like that. `case 0: if (chance < config.cleanerSpawnChance)`